packages/core/src/integrations/mcp-server/sessionExtraction.ts

```diff
@@ -0,0 +1,196 @@
+/**
+ * Session and party info extraction functions for MCP server instrumentation
+ *
+ * Handles extraction of client/server info and session data from MCP messages.
+ */
+
+import {
+  CLIENT_ADDRESS_ATTRIBUTE,
+  CLIENT_PORT_ATTRIBUTE,
+  MCP_PROTOCOL_VERSION_ATTRIBUTE,
+  MCP_SERVER_NAME_ATTRIBUTE,
+  MCP_SERVER_TITLE_ATTRIBUTE,
+  MCP_SERVER_VERSION_ATTRIBUTE,
+  MCP_SESSION_ID_ATTRIBUTE,
+  MCP_TRANSPORT_ATTRIBUTE,
+  NETWORK_PROTOCOL_VERSION_ATTRIBUTE,
+  NETWORK_TRANSPORT_ATTRIBUTE,
+} from './attributes';
+import {
+  getClientInfoForTransport,
+  getProtocolVersionForTransport,
+  getSessionDataForTransport,
+} from './sessionManagement';
+import type { ExtraHandlerData, JsonRpcRequest, MCPTransport, PartyInfo, SessionData } from './types';
+
+/**
+ * Extracts and validates PartyInfo from an unknown object
+ * @param obj - Unknown object that might contain party info
+ * @returns Validated PartyInfo object with only string properties
+ */
+function extractPartyInfo(obj: unknown): PartyInfo {
+  const partyInfo: PartyInfo = {};
+
+  if (obj && typeof obj === 'object' && obj !== null) {
+    const source = obj as Record<string, unknown>;
+    if (typeof source.name === 'string') partyInfo.name = source.name;
+    if (typeof source.title === 'string') partyInfo.title = source.title;
+    if (typeof source.version === 'string') partyInfo.version = source.version;
```

Comment:
l: I prefer if avoid inline conditionals as it makes future diffs easier to read. With the version below, intent when git blaming when we change the condition vs. changing the setter is more clear 

Ditto everywhere else in the code that this happens.

```suggestion
    if (typeof source.name === 'string') {
      partyInfo.name = source.name;
    }
    if (typeof source.title === 'string') {
      partyInfo.title = source.title;
    }
    if (typeof source.version === 'string') {
      partyInfo.version = source.version;
    }
```

Reply:
i was doing it inline to avoid getting hit by our 300 code lines limit + it's my style in python but it makes so much sense the blame reason. fixed!